satpy/modifiers/angles.py

```diff
@@ -0,0 +1,266 @@
+        return f"{self._func.__name__}_v{cache_version}" + "_{}_" + f"{arg_hash}.zarr"
+
+    def __call__(self, *args, cache_dir: Optional[str] = None) -> Any:
+        """Call the decorated function."""
+        new_args = self._sanitize_args_func(*args) if self._sanitize_args_func is not None else args
+        arg_hash = _hash_args(*new_args)
+        should_cache, cache_dir = self._get_should_cache_and_cache_dir(new_args, cache_dir)
+        zarr_fn = self._zarr_pattern(arg_hash)
+        zarr_format = os.path.join(cache_dir, zarr_fn)
+        zarr_paths = glob(zarr_format.format("*"))
+        if not should_cache or not zarr_paths:
+            # use sanitized arguments if we are caching, otherwise use original arguments
+            args = new_args if should_cache else args
+            res = self._func(*args)
+            if should_cache and not zarr_paths:
+                self._cache_results(res, zarr_format)
+        # if we did any caching, let's load from the zarr files
+        if should_cache:
+            # re-calculate the cached paths
+            zarr_paths = glob(zarr_format.format("*"))
+            if not zarr_paths:
+                raise RuntimeError("Data was cached to disk but no files were found")
+            res = tuple(da.from_zarr(zarr_path) for zarr_path in zarr_paths)
+        return res
+
+    def _get_should_cache_and_cache_dir(self, args, cache_dir):
+        should_cache = satpy.config.get(self._cache_config_key, False)
+        can_cache = not any(isinstance(arg, self._uncacheable_arg_types) for arg in args)
+        should_cache = should_cache and can_cache
+        if cache_dir is None:
+            cache_dir = satpy.config.get("cache_dir")
+        if cache_dir is None:
+            should_cache = False
+        return should_cache, cache_dir
+
+    def _cache_results(self, res, zarr_format):
+        os.makedirs(os.path.dirname(zarr_format), exist_ok=True)
+        new_res = []
+        for idx, sub_res in enumerate(res):
+            zarr_path = zarr_format.format(idx)
+            # See https://github.com/dask/dask/issues/8380
+            with dask.config.set({"optimization.fuse.active": False}):
+                new_sub_res = sub_res.to_zarr(zarr_path,
+                                              return_stored=True,
+                                              compute=False)
+            new_res.append(new_sub_res)
+        # actually compute the storage to zarr
+        da.compute(new_res)
+
+
+def cache_to_zarr(
+        cache_config_key: str,
+        uncacheable_arg_types=(SwathDefinition, xr.DataArray, da.Array),
+        sanitize_args_func: Callable = None,
+) -> Callable:
+    """Decorate a function and cache the results as a zarr array on disk.
+
+    Note: Only the dask array is cached. Currently the metadata and coordinate
+    information is not stored.
+
+    """
+    # TODO: Consider cache management (LRU)
```

Comment:
I think the docstring should mention what the behaviour is now (no limit caching, right?)

Reply:
Agreed. And also mention something how it is determined if the existing cache is usable or not as there's no metadata included to check the cache validity.